### PR TITLE
Allow CALL functions without RETURN clause

### PR DESCRIPTION
**Pull request by royi-luo**
# Description

We have some "void" CALL functions such as `clear_warnings()`, this removes the need for these functions to return any values.

To do this, we modify the grammar to break `oC_ReadingClause` into `oC_MandatoryReturnReadingClause` (for clauses like `MATCH` and `LOAD_FROM`) and `oC_OptionalReturnReadingClause` (for CALL clauses). For optional return reading clauses, the check for `RETURN` is moved to the binding phase and only throws if the `CALL` clause returns non-empty output.

Fixes https://github.com/kuzudb/kuzu/issues/4250

# Contributor agreement

- [ ] I have read and agree to the [Contributor Agreement](https://github.com/kuzudb/kuzu/blob/master/CLA.md).

### CODE REVIEW

File: scripts/antlr4/Cypher.g4 [hunk 1]
```diff
@@ -478,17 +481,33 @@ oC_UpdatingClause
...
 kU_InQueryCall
+    : kU_InQueryTableFunctionCall
+        | kU_InQueryBuiltInCall
+        ;
+
+kU_InQueryTableFunctionCall
+    : ( kU_ProjectGraph SP? )? CALL SP oC_TableFunctionCall (SP? oC_Where)? ;
+
+kU_InQueryBuiltInCall
     : ( kU_ProjectGraph SP? )? CALL SP oC_FunctionInvocation (SP? oC_Where)? ;
```
Comment (royi-luo) [inline scripts/antlr4/Cypher.g4:511]: Is `CALL COUNT(*)` or `CALL CAST` a thing or can I remove `kU_InQueryBuiltInCall` altogether?

**Review by andyfengHKU — CHANGES_REQUESTED** [review]
hmmmm, this is wrong way to handle `CALL clear_warning()`.

Don't compile it as a `SingleQuery` or `TableFunction`.

Compile it as as a `StandaloneCall`

**Comment by royi-luo** [comment]
> hmmmm, this is wrong way to handle `CALL clear_warning()`.
> 
> Don't compile it as a `SingleQuery` or `TableFunction`.
> 
> Compile it as a `StandaloneCall`.
> 
> I suggest let @acquamarin take over or mentor.

I see, will update and let @acquamarin know when I'm done

**Comment by andyfengHKU** [comment]
> > hmmmm, this is wrong way to handle `CALL clear_warning()`.
> > Don't compile it as a `SingleQuery` or `TableFunction`.
> > Compile it as a `StandaloneCall`.
> > I suggest let @acquamarin take over or mentor.
> 
> I see, will update and let @acquamarin know when I'm done

Talk to @acquamarin before u start. I don't know if he plans to take over or not.

**Comment by royi-luo** [comment]
Closing as @acquamarin is taking this over.

**Pull request closed without merging**